sphinx/builders/linkcheck.py

```diff
@@ -387,7 +387,7 @@ def __init__(
         )
         self.check_anchors: bool = config.linkcheck_anchors
         self.allowed_redirects: dict[re.Pattern[str], re.Pattern[str]]
-        self.allowed_redirects = config.linkcheck_allowed_redirects or {}
+        self.allowed_redirects = config.linkcheck_allowed_redirects
```

Comment:
In hindsight, I think that -- alongside a [corresponding undo](https://github.com/sphinx-doc/sphinx/pull/13483#pullrequestreview-2887530464) in the `_allowed_redirect` helper method -- we do not need to change this part of the code.  So this can/should be restored to:

```suggestion
        self.allowed_redirects = config.linkcheck_allowed_redirects or {}
```

Edit: add bidirectional undo-patch hyperlink

Reply:
(done)